drivers/winc1500/winc1500_netdev.c

```diff
@@ -0,0 +1,250 @@
+
+    dev->state = 0 | WINC1500_STATE_INIT;
+    dev->state |= WINC1500_STATE_IDLE;
+
+    _unlock_bus(dev);
+
+#ifdef MODULE_NETSTATS_L2
+    memset(&netdev->stats, 0, sizeof(netstats_t));
+#endif
+    return 0;
+}
+
+
+static int _send(netdev_t *netdev, const struct iovec *vector, unsigned count) {
+    winc1500_t * dev = (winc1500_t *) netdev;
+    _lock_bus(dev);
+
+    /* Send packet */
+    size_t len = 0;
+    for (int i = 0; i < count; i++) {
+        m2m_wifi_send_ethernet_pkt(vector[i].iov_base, vector[i].iov_len);
+        len += vector[i].iov_len;
+        /* TODO: what if the module's buffer full? */
+    }
+
+#ifdef MODULE_NETSTATS_L2
+    netdev->stats.tx_bytes += len;
+#endif
+
+    _unlock_bus(dev);
+    return len;
+}
+
+
+static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
+{
+    winc1500_t * dev = (winc1500_t *) netdev;
+
+    (void)info;
+    _lock_bus(dev);
+
+    int size;
+    uint16_t data_size;
+    uint8_t rx_done;
+    winc1500_event_info_t event_info;
+
+    if (dev->rx_size == -1) {
+        if (hif_receive(dev->rx_addr, (uint8_t *) &event_info,
+                    sizeof(event_info.recv_pkt), 0) == M2M_SUCCESS) {
+            dev->rx_offset = event_info.recv_pkt.u16PktOffset;
+            /* TODO: Subtrack 4 byte checksum?? There is no infomation on WINC1500 */
+            dev->rx_rem_size = event_info.recv_pkt.u16PktSz;
+            dev->rx_size = event_info.recv_pkt.u16PktSz;
+        }
+    }
+    size = dev->rx_size;
+    if (buf != NULL) {
+#ifdef MODULE_NETSTATS_L2
+        netdev->stats.rx_count++;
+        netdev->stats.rx_bytes += len;
+#endif
+        /* Firstly, retrive MAC Header part */
+        data_size = 14; /* Dest MAC(6) + Source MAC(6) + EtherType(2) */
+        rx_done = 0;
+        if(hif_receive(dev->rx_addr + dev->rx_offset, buf, data_size, rx_done) == M2M_SUCCESS) {
+            dev->rx_rem_size -= data_size;
+            dev->rx_offset += data_size + 2; /** The WINC1500 module have
+                                            strange extra 2 bytes. Skip them */
+            buf += data_size;
+        }
+
+        /* Then get the rest of it */
+        do {
+            rx_done = 1;
+            if(dev->rx_rem_size > len) {
+                rx_done = 0;
+                data_size = len;
+            }
+            else {
+                data_size = dev->rx_rem_size;
+            }
+
+            if(hif_receive(dev->rx_addr + dev->rx_offset, buf, data_size, rx_done) == M2M_SUCCESS) {
+                dev->rx_rem_size -= data_size;
+                dev->rx_offset += data_size;
+                buf += data_size;
+            }
+            else {
+                break;
+            }
+        } while (dev->rx_rem_size > 0);
+        dev->rx_size = -1;
+    }
+    _unlock_bus(dev);
+    return size;
+}
+
+
+static int _get(netdev_t *netdev, netopt_t opt, void *value, size_t max_len)
+{
+    /* winc1500_t *dev = (winc1500_t *) netdev; */
```

Comment:
This is actually the correct way normally

Reply:
Indeed. I commented out here because `winc1500_t *dev` is not used in that function. I think `extern` global parameter problem doesn't exist at least in `winc1500_netdev.c`.